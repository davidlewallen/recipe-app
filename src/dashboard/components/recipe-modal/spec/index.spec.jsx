import React from 'react';
import { shallow } from 'enzyme';

import RecipeModal from '../components';

describe('RecipeModal component', () => {
  const props = {
    showModal: false,
    handleModalClose: jest.fn(),
    selectedRecipe: {
      title: 'test',
      ingredients: ['1'],
      instructions: ['1'],
      totalTime: '1 hour',
      url: {
        href: 'randomurl.com',
      },
    },
    deleteRecipe: jest.fn(),
  };

  it('should match snapshot', () => {
    const component = shallow(<RecipeModal {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should match snapshot with modal open', () => {
    const component = shallow(<RecipeModal {...props} showModal />);
    expect(component).toMatchSnapshot();
  });

  it('should render', () => {
    const component = shallow(<RecipeModal {...props} />);
    expect(component).toBeTruthy();
  });

  it('should display total time', () => {
    const component = shallow(<RecipeModal {...props} />);
    const time = component.find('span.total-time');

    expect(time.text()).toBe(`Total Time: ${props.selectedRecipe.totalTime}`);
  });

  it('should display "n/a" if total time isnt provided', () => {
    const modifiedProps = props;
    delete modifiedProps.selectedRecipe.totalTime;

    const component = shallow(<RecipeModal {...modifiedProps} />);
    const time = component.find('span.total-time');
    expect(time.text()).toBe('Total Time: n/a');
  });

  it('should render multiple ingredients', () => {
    const modifiedProps = props;
    modifiedProps.selectedRecipe.ingredients.push('1');

    const component = shallow(<RecipeModal {...modifiedProps} />);
    const ingredients = component.find('div.ingredient');
    expect(ingredients.length).toBe(2);
  });

  it('should render multiple instructions', () => {
    const modifiedProps = props;
    modifiedProps.selectedRecipe.instructions.push('1');

    const component = shallow(<RecipeModal {...modifiedProps} />);
    const instructions = component.find('div.instruction');
    expect(instructions.length).toBe(2);
  });

  it('should render a title', () => {
    const component = shallow(<RecipeModal {...props} />);
    const title = component.find('.title').childAt(0);
    expect(title.text()).toBe(props.selectedRecipe.title);
  });

  it('should link to the original recipe website', () => {
    const component = shallow(<RecipeModal {...props} />);
    const link = component.find('a.recipe-link');
    expect(link.props().href).toBe(props.selectedRecipe.url.href);
  });

  it('should call deleteRecipe with recipe id when delete button is pressed', () => {
    const component = shallow(<RecipeModal {...props} />);
    const button = component.find('Button');
    expect(props.deleteRecipe).toHaveBeenCalledTimes(0);
    button.simulate('click');
    expect(props.deleteRecipe).toHaveBeenCalledWith(props.selectedRecipe._id);
  });
});
