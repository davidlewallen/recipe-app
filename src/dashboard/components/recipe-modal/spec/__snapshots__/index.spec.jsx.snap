// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeModal component should match snapshot 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  className="recipe-modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[MockFunction]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <ModalHeader
    bsClass="modal-header"
    className="header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      className="title"
      componentClass="h4"
    >
      <span>
        test
      </span>
      <span
        className="total-time"
      >
        Total Time: 1 hour
      </span>
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    className="body"
    componentClass="div"
  >
    <div
      className="ingredient-title"
    >
      Ingredients:
    </div>
    <div
      className="ingredient"
      key="a347ccac-5c74-51c8-b109-9dea71b3111e"
    >
      - 1
    </div>
    <div
      className="instruction-title"
    >
      Instructions:
    </div>
    <div
      className="instruction"
      key="a347ccac-5c74-51c8-b109-9dea71b3111e"
    >
      1
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <a
      className="recipe-link"
      href="randomurl.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Click here to view original recipe
    </a>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsSize="small"
      bsStyle="danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Delete
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`RecipeModal component should match snapshot with modal open 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  className="recipe-modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[MockFunction]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    className="header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      className="title"
      componentClass="h4"
    >
      <span>
        test
      </span>
      <span
        className="total-time"
      >
        Total Time: 1 hour
      </span>
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    className="body"
    componentClass="div"
  >
    <div
      className="ingredient-title"
    >
      Ingredients:
    </div>
    <div
      className="ingredient"
      key="a347ccac-5c74-51c8-b109-9dea71b3111e"
    >
      - 1
    </div>
    <div
      className="instruction-title"
    >
      Instructions:
    </div>
    <div
      className="instruction"
      key="a347ccac-5c74-51c8-b109-9dea71b3111e"
    >
      1
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <a
      className="recipe-link"
      href="randomurl.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Click here to view original recipe
    </a>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsSize="small"
      bsStyle="danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Delete
    </Button>
  </ModalFooter>
</Modal>
`;
